File: starlore-front/src/api/project.ts
import request from '@/utils/request'

export interface Project {
    id: number
    name: string
    description: string | null
    url: string | null
    image: string | null
    sortOrder: number | null
    createdAt: string
    updatedAt: string
}

export interface ProjectParams {
    name: string
    description?: string
    url?: string
    image?: string
    sortOrder?: number
}

export function listProjectsService() {
    return request.get<never, { data: Project[] }>('/projects')
}

export function createProjectService(params: ProjectParams) {
    return request.post<never, { data: Project; message: string }>('/projects', params)
}

export function updateProjectService(id: number, params: ProjectParams) {
    return request.put<never, { data: Project; message: string }>(`/projects/${id}`, params)
}

export function deleteProjectService(id: number) {
    return request.delete<never, { message: string }>(`/projects/${id}`)
}

File: starlore-py/app/schemas/project.py
"""项目相关 Schema。"""

from datetime import datetime

from pydantic import BaseModel


class ProjectRequest(BaseModel):
    name: str | None = None
    description: str | None = None
    url: str | None = None
    image: str | None = None
    sortOrder: int | None = None


class ProjectInfo(BaseModel):
    id: int
    name: str
    description: str | None = None
    url: str | None = None
    image: str | None = None
    sortOrder: int | None = None
    createdAt: datetime | None = None
    updatedAt: datetime | None = None


class ProjectResponse(BaseModel):
    data: ProjectInfo
    message: str

